### PR TITLE
Fixes multiple PRs raised by Test team

### LINKED ISSUE #104: salt execution module junos.install_config with mode dynamic throws exception RpcError: RpcError(severity: error, bad_element: None, message: error: statement creation failed: version
### Description of Issue/Question

```
set policy-options policy-statement test term test from route-filter 10.0.1.0/24 address-mask 255.0.255.0
set policy-options policy-statement test then accept

root@nms5-salt-master-b:/srv/salt# cat config_dynamic.xml 
<policy-options>
                <policy-statement>
                    <name>test</name>
                    <term>
                        <name>test</name>
                        <from>
                            <route-filter>
                                <address>10.0.1.0/24</address>
                                <address-mask>255.0.255.0</address-mask>
                            </route-filter>
                        </from>
                    </term>
                    <then>
                        <accept/>
                    </then>
                </policy-statement>
            </policy-options>




root@nms5-salt-master-b:/srv/salt# salt "vmx1" junos.install_config salt://config_dynamic.set mode=dynamic comment="commit dynamic mode" format=set
[WARNING ] /usr/local/lib/python2.7/site-packages/salt/payload.py:149: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  ret = msgpack.loads(msg, use_list=True, ext_hook=ext_type_decoder, encoding=encoding)

vmx1:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/local/lib/python2.7/site-packages/salt/minion.py", line 1660, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "/usr/local/lib/python2.7/site-packages/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/salt/modules/junos.py", line 99, in wrapper
        return function(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/salt/modules/junos.py", line 935, in install_config
        config_diff = cu.diff()
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/utils/config.py", line 228, in diff
        compare='rollback', rollback=str(rb_id), format='text'
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/rpcmeta.py", line 345, in _exec_rpc
        return self._junos.execute(rpc, **dec_args)
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/decorators.py", line 76, in wrapper
        return function(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/decorators.py", line 31, in wrapper
        return function(*args, **kwargs)
      File "/usr/local/lib/python2.7/site-packages/jnpr/junos/device.py", line 794, in execute
        errs=ex)
    RpcError: RpcError(severity: error, bad_element: None, message: error: statement creation failed: version
    error: /config/juniper.conf:5:(0) statement creation failed
    error: /config/juniper.conf:100:(9) error recovery ignores input until this point
    error: /config/juniper.conf:104:(9) error recovery ignores input until this point
    error: /config/juniper.conf:124:(9) error recovery ignores input until this point
    error: /config/juniper.conf:135:(9) error recovery ignores input until this point
    error: /config/juniper.conf:210:(9) error recovery ignores input until this point
    warning: mgd: statement must contain additional statements
    error: /config/juniper.conf:212:(0) statement creation failed
    error: /config/juniper.conf:216:(9) error recovery ignores input until this point
    error: /config/juniper.conf:225:(9) error recovery ignores input until this point
    warning: mgd: statement must contain additional statements
    error: /config/juniper.conf:227:(0) statement creation failed
    error: /config/juniper.conf:231:(9) error recovery ignores input until this point
    error: /config/juniper.conf:240:(9) error recovery ignores input until this point
    warning: mgd: statement must contain additional statements
    error: /config/juniper.conf:243:(0) statement creation failed
    error: /config/juniper.conf:265:(1) error recovery ignores input until this point
    error: /config/juniper.conf:275:(1) error recovery ignores input until this point)
ERROR: Minions returned with non-zero exit code
root@nms5-salt-master-b:/srv/salt# salt "vmx1" junos.install_config salt://config_dynamic.set mode=dynamic comment="commit dynamic mode"
```

### PR DESCRIPTION
**Pull request by rsmekala**
### What does this PR do?

Fixes #104

### CODE REVIEW

**Comment by rsmekala** [comment]
Fixes #112

**Comment by rsmekala** [comment]
Fixes #109

**Comment by rsmekala** [comment]
Fixes #113, #114

**Comment by rsmekala** [comment]
Fixes #115

File: salt/modules/junos.py [hunk 1]
```diff
@@ -939,29 +945,31 @@ def install_config(path=None, **kwargs):
...
+                        'Commit check threw the following exception: "{0}"'\
+                        .format(exception)
 
-                ret['out'] = False
-                return ret
+                    ret['out'] = False
+                    return ret
 
-            if check and not test:
+            if db_mode == 'dynamic' or check and not test:
```
Comment (vnitinv) [inline salt/modules/junos.py]: can you just use default value for check and then we wont need this line change

File: salt/modules/junos.py [hunk 2]
```diff
@@ -1105,22 +1128,29 @@ def install_os(path=None, **kwargs):
...
+        # To handle invalid image path scenario
+        try:
+            image_cached_path = salt.utils.files.mkstemp()
+            __salt__['cp.get_file'](path, image_cached_path)
 
-        if not os.path.isfile(image_cached_path):
-            ret['message'] = 'Invalid image path.'
-            ret['out'] = False
-            return ret
+            if not os.path.isfile(image_cached_path):
```
Comment (vnitinv) [inline salt/modules/junos.py:1139]: check if this is Tab?

File: salt/modules/junos.py [hunk 3]
```diff
@@ -1510,7 +1555,7 @@ def get_table(table, table_file, path=None, target=None, key=None, key_items=Non
     except Exception as err:
         ret['message'] = 'Uncaught exception - please report: {0}'.format(
             str(err))
-        traceback.print_exc()
+
```
Comment (vnitinv) [inline salt/modules/junos.py]: revert the removed line back

**Review by vnitinv — CHANGES_REQUESTED** [review]
please do the requested changes.

File: salt/modules/junos.py (hunk 1 above)
Comment (rsmekala) [reply salt/modules/junos.py]: `commit_check` returns `True` if commit_check succeeds, else raises an exception.

Link: https://github.com/Juniper/py-junos-eznc/blob/dabef6dc29af8017d5eb8ae4be70daa7a42b77d6/lib/jnpr/junos/utils/config.py#L175

File: salt/modules/junos.py (hunk 2 above)
Comment (rsmekala) [reply salt/modules/junos.py:1139]: Verified no Tab.

File: salt/modules/junos.py (hunk 3 above)
Comment (rsmekala) [reply salt/modules/junos.py]: Reverted it back

**Comment by rsmekala** [comment]
@vnitinv Also updated the comment as requested.

**Review by vnitinv — APPROVED** [review]
looks good

**Pull request merged**